src/openrct2/actions/GuestSetNameAction.hpp

```diff
@@ -0,0 +1,127 @@
+#pragma region Copyright (c) 2014-2017 OpenRCT2 Developers
+/*****************************************************************************
+* OpenRCT2, an open source clone of Roller Coaster Tycoon 2.
+*
+* OpenRCT2 is the work of many authors, a full list can be found in contributors.md
+* For more information, visit https://github.com/OpenRCT2/OpenRCT2
+*
+* OpenRCT2 is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* A full copy of the GNU General Public License can be found in licence.txt
+*****************************************************************************/
+#pragma endregion
+
+#pragma once
+
+#include "../core/MemoryStream.h"
+#include "../localisation/string_ids.h"
+#include "GameAction.h"
+
+#include "../cheats.h"
+#include "../interface/window.h"
+#include "../localisation/localisation.h"
+#include "../world/park.h"
+
+struct GuestSetNameAction : public GameActionBase<GAME_COMMAND_SET_GUEST_NAME, GameActionResult>
+{
+private:
+    uint16 _spriteIndex;
+    std::string _name;
+
+public:
+    GuestSetNameAction() {}
+    GuestSetNameAction(uint16 spriteIndex, const std::string& name)
+        : _spriteIndex(spriteIndex),
+        _name(name)
+    {
+    }
+
+    uint16 GetActionFlags() const override
+    {
+        return GameAction::GetActionFlags() | GA_FLAGS::ALLOW_WHILE_PAUSED;
+    }
+
+    void Serialise(DataSerialiser& stream) override
+    {
+        GameAction::Serialise(stream);
+
+        stream << _spriteIndex << _name;
+    }
+
+    GameActionResult::Ptr Query() const override
+    {
+        if (_spriteIndex >= MAX_SPRITES)
+        {
+            return std::make_unique<GameActionResult>(GA_ERROR::INVALID_PARAMETERS, STR_NONE);
+        }
+
+        if (_name.empty())
+        {
+            return std::make_unique<GameActionResult>(GA_ERROR::INVALID_PARAMETERS, STR_ERR_INVALID_NAME_FOR_GUEST);
+        }
+
+        rct_peep *peep = GET_PEEP(_spriteIndex);
+        if (peep->type != PEEP_TYPE_GUEST)
+        {
+            log_warning("Invalid game command for sprite %u", _spriteIndex);
+            return std::make_unique<GameActionResult>(GA_ERROR::INVALID_PARAMETERS, STR_NONE);
+        }
+
+        rct_string_id newUserStringId = user_string_allocate(USER_STRING_HIGH_ID_NUMBER | USER_STRING_DUPLICATION_PERMITTED, _name.c_str());
+        if (newUserStringId == 0)
+        {
+            // TODO: Probably exhausted, introduce new error.
+            return std::make_unique<GameActionResult>(GA_ERROR::UNKNOWN, STR_NONE);
```

Comment:
Looks like this is communicated via `gGameCommandErrorText` can you set the error to have value of that variable?